packages/openapi3-parser/lib/parser/oas/parseComponentsObject.js

```diff
@@ -117,24 +131,23 @@ function parseComponentsObject(context, element) {
    * @returns ParseResult<ObjectElement>
    * @private
    */
-  const parseComponentObjectMember = (parser) => {
+  const parseComponentObjectMember = R.curry((parser, member) => {
     const parseMember = parseComponentMember(context, parser);
+    const parseMemberOrRef = m => parseReference(member.key.toValue(), () => parseMember(m), context, m.value, false, true);
 
-    return member => pipeParseResult(context.namespace,
+    return pipeParseResult(context.namespace,
       validateIsObject,
-      R.compose(parseObject(context, name, parseMember), getValue),
+      R.compose(parseObject(context, name, parseMemberOrRef), getValue),
       (object) => {
         const contextMember = context.state.components.getMember(member.key.toValue());
 
         if (contextMember) {
           contextMember.value = object;
-        } else {
```

Comment:
For review, this branch becomes dead code due to the first commit we only ever have the first case. This was evident in code coverage going down due to this line no longer being called/tested.